src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -94,10 +94,23 @@ public static AppiumDriverLocalService buildService(AppiumServiceBuilder builder
     }
 
     public AppiumDriverLocalService withBasePath(String basePath) {
-        this.basePath = basePath;
+        this.basePath = sanitizeBasePath(basePath);
         return this;
     }
 
+    @SneakyThrows private static String sanitizeBasePath(String basePath) {
+        if (null == basePath) {
+            LOG.warn("Base Path cannot be NULL -- ignoring the basepath configuration");
+            return null;
+        }
+        basePath = basePath.trim();
+        if (basePath.isBlank() || basePath.isEmpty()) {
+            LOG.warn("Base Path cannot be Empty or Blank -- ignoring the basepath configuration");
+            return null;
+        }
+        return (basePath.endsWith("/") ? basePath : basePath + "/");
```

Comment:
redundant parenthesis
```suggestion
        return basePath.endsWith("/") ? basePath : basePath + "/";
```